src/api/app/components/buildresult_status_link_component.rb

```diff
@@ -0,0 +1,27 @@
+class BuildresultStatusLinkComponent < ApplicationComponent
+  def initialize(repository:, architecture:, project:, package_name:, build_status:, build_details:)
+    super
+
+    @repository = repository
+    @architecture = architecture
+    @project = project
+    @package_name = package_name
+    @build_details = build_details
+    @build_status = build_status
+  end
+
+  def show_link?
+    ['-', 'unresolvable', 'blocked', 'excluded', 'scheduled'].exclude?(@build_status)
+  end
+
+  def css_class
+    # special case for scheduled jobs with constraints limiting the workers a lot
+    return 'text-warning' if @build_status == 'scheduled' && @build_details.present?
+
+    "build-state-#{@build_status}" if @build_status.present?
+  end
+
+  def span_id
+    "id-#{@package_name}_#{@repository}_#{@architecture}"
```

Comment:
I didn't transferred the usage of `valid_xml_id` from the helper, I think this is not required since the escaped character are not allowed anyway in the package name etc.